Motivation and Mindset: Add image for /stuck diagram

nikitarevenco opened the pull request:
<!-- Thank you for taking the time to contribute to The Odin Project. In order to get this pull request (PR) merged in a reasonable amount of time, you must complete this entire template. -->

## Because
<!-- Summarize the purpose or reasons for this PR, e.g. what problem it solves or what benefit it provides. -->
useful diagram for when stuck

## This PR
<!-- A bullet point list of one or more items describing the specific changes. -->
Adds the image for /stuck diagram

## Issue
<!--
If this PR closes an open issue in this repo, replace the XXXXX below with the issue number, e.g. Closes #2013.

If this PR closes an open issue in another TOP repo, replace the #XXXXX with the URL of the issue, e.g. Closes https://github.com/TheOdinProject/curriculum/issues/XXXXX

If this PR does not close, but is related to another issue or PR, you can link it as above without the 'Closes' keyword, e.g. 'Related to #2013'.

_Note:_ any pull request created for an issue that already has someone else assigned **will be closed without review**.
-->
Closes #27523

## Additional Information
<!-- Any other information about this PR, such as a link to a Discord discussion. -->


## Pull Request Requirements
<!-- Replace the whitespace between the square brackets with an 'x', e.g. [x]. After you create the PR, they will become checkboxes that you can click on. -->
-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

JoshDevHub reviewed (commented):
Looks good @nikitarevenco 

Would you be okay fixing the lint errors in this lesson while you're here? You can ignore the heading structure one because that'll be a bigger change, but the ones around blank lines and list styles could be easy to clean up if you want.

nikitarevenco commented:
> Looks good @nikitarevenco
> 
> Would you be okay fixing the lint errors in this lesson while you're here? You can ignore the heading structure one because that'll be a bigger change, but the ones around blank lines and list styles could be easy to clean up if you want.

Alright, I've fixed the issues mostly but I have a few questions

This lint issue says I should remove the whitespace to the left of the unordered list dashes `-`, however if I do that, then the list structure will change, I am pretty sure the lists need to be nested? I checked with the markdown preview tool and that does appear to be the case.
<img src="https://github.com/TheOdinProject/curriculum/assets/154856872/141092fd-8302-4644-8e2e-b7cfd65b0a77" width="30%" height="auto">

Also, since the additional resources shouldn't have nested lists, I've removed the 3 extra typing tool options and left a single one only. I personally don't think it makes a difference if we recommend 1 tool vs 4 tools, since all of them are pretty much the same. What do you think about this? Thinking about other ways this problem could be solved without removing the mention of those 3 extra typing speed tools

nikitarevenco commented on foundations/introduction/motivation_and_mindset.md:
```diff
@@ -172,12 +178,7 @@ This section contains helpful links to related content. It isn't required, so co
-  - [typing.com](https://www.typing.com/)
-  - [TypingClub.com](https://www.typingclub.com)
-  - [Monkeytype](https://monkeytype.com/)
+- If you find your typing speed is holding you back, you can improve your typing skills by utilizing the [Monkeytype](https://monkeytype.com/) keyword trainer for at least 5 minutes every day.
```
> ```suggestion
> - If you find your typing speed is holding you back, you can improve your typing skills by utilizing keyword trainers for at least 5 minutes every day. Some excellent options include [Monkeytype](https://monkeytype.com/), [typing.com](https://www.typing.com/), [TypingClub.com](https://www.typingclub.com) and [Monkeytype](https://monkeytype.com/).
> ```
> 
> What about adding all of them inline?

This is a possibility, but one might have to ask why even mention 4 sites for typing speed? The main goal is to represent the idea. Use a typing speed site to improve your typing speed. Here is an example. The learner can research another one if they don't like the current one, but really they're all the same

cakegod replied:
It provides additional base options. Although I am indifferent towards either solution.

JoshDevHub replied:
I like adding the different options inline, so +1 for cake's idea here.

I like MonkeyType, but some of the other given options are a bit more formal, which might suit certain users' needs better. Like if you don't know touch typing at all, you may want something that's more of a "course" that walks through hand positioning and what fingers to use with each key, etc., and those other sites have that kind of thing going on.

nikitarevenco replied:
> I like adding the different options inline, so +1 for cake's idea here.
> 
> I like MonkeyType, but some of the other given options are a bit more formal, which might suit certain users' needs better. Like if you don't know touch typing at all, you may want something that's more of a "course" that walks through hand positioning and what fingers to use with each key, etc., and those other sites have that kind of thing going on.

Oh yeah, in that case I definitely agree!

JoshDevHub commented:
> This lint issue says I should remove the whitespace to the left of the unordered list dashes -, however if I do that, then the list structure will change, I am pretty sure the lists need to be nested? I checked with the markdown preview tool and that does appear to be the case.

Try formatting these the same way they're done in the [LAYOUT_STYLE_GUIDE](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md#nested-lists)
```md
<!-- bad -->
1. List item

  - unordered item 1
  - unordered item 2
  
<!-- good -->
1. List item
   - unordered item 1
   - unordered item 2
```
Notice the absence of a blank line between the OL item and the UL items, as well as the three space indentation for the nested unordered items.

JoshDevHub commented on foundations/introduction/motivation_and_mindset.md:
```diff
@@ -161,6 +165,7 @@ Learning any new skill is a journey full of speed bumps and obstacles to be over
   - [Believe you can get better](https://www.ted.com/talks/carol_dweck_the_power_of_believing_that_you_can_improve)
```
The linter error here is confusing, but it should be fixed if you indent these bullets one more step to the right.

JoshDevHub reviewed (changes requested):
Just one little change with the bullets and this should be good to merge.

JoshDevHub reviewed (approved):
Looks great :rocket:
